sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/MultimapSideInput.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.fn.harness.state;
+
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.function.Supplier;
+import org.apache.beam.fn.harness.stream.DataStreams;
+import org.apache.beam.model.fnexecution.v1.BeamFnApi.StateRequest;
+import org.apache.beam.sdk.coders.Coder;
+
+/**
+ * An implementation of a multimap side input that utilizes the Beam Fn State API to fetch values.
+ *
+ * <p>TODO: Support block level caching and prefetch.
+ */
+public class MultimapSideInput<K, V> {
+  private final BeamFnStateClient beamFnStateClient;
```

Comment:
<!--thread_id:cc_145816474_t; commit:d1c0ffde59d0b17126deecb840cbeec36719d7a6; resolved:1-->
<!--section:context-quote-->
> **lukecwik** wrote:
> No, its part of the partial state request passed into this by the FnApiDoFnRunner, see FnApiDoFnRunner.getMultimapSideInputKey.

<!--section:body-->
Got it